data.tf

```diff
@@ -0,0 +1,74 @@
+locals {
+  // yamlencode is producing quoted keys yaml
+  // this is a workaround while tf folks improve the yamlencode function
+  // ref: https://github.com/hashicorp/terraform/issues/23322
+  yaml           = replace(yamlencode(local.cloudinit_config), "\"", "")
+  cloudinit_tmpl = templatefile("${path.module}/templates/cloudinit.tftpl", { "yaml" = local.yaml })
+  cloudinit_config = {
```

Comment:
would this approach enable a merge functionality to allow the caller to overwrite/extend cloudinit configuration?